libs/utils/utils.js

```diff
@@ -725,6 +774,42 @@ export function decorateLinks(el) {
     if (a.href.includes(copyEvent)) {
       decorateCopyLink(a, copyEvent);
     }
+    const branchQuickLink = 'app.link';
+    const ecidCheck = getMetadata('quick-link-ecid'); // on | off | null
+    const loaderCheck = getMetadata('quick-link-loader'); // progress-circle | progress-bar | off | null
+
+    if (a.href.includes(branchQuickLink) && ecidCheck === 'on') {
+      const getConsentStatus = () => {
+        const cookieGrp = window.adobePrivacy?.activeCookieGroups();
+        return cookieGrp?.includes('C0002') && cookieGrp?.includes('C0004');
+      };
+
+      const waitForConsent = new Promise((resolve) => {
+      // case: not first encounter
+        if (window.cookieConsent !== undefined) {
+          resolve(window.cookieConsent);
+        } else {
+          if (window.adobePrivacy) resolve(getConsentStatus());
+          window.addEventListener('adobePrivacy:PrivacyConsent', () => {
+            window.cookieConsent = getConsentStatus();
+            resolve(window.cookieConsent);
+          });
+        }
+      });
+
+      a.addEventListener('click', async (e) => {
+        e.preventDefault();
+        let pb;
+        if (loaderCheck === 'progress-circle') addCircleLoader(a);
+        else if (loaderCheck === 'progress-bar') pb = addBarLoader(a);
+        const hasConsent = await waitForConsent;
+        if (hasConsent) {
+          if (loaderCheck === 'progress-bar') removeBarLoader(pb, a);
+          else if (loaderCheck === 'progress-circle') removeCircleLoader(a);
+        }
+        decorateQuickLink(a, hasConsent);
+      });
```

Comment:
This is a bit hard to read without the standard brackets that we use for `if` `else if` etc. I think you may also be able to flatten with early returns.